airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -101,6 +115,10 @@ def get_conn(self):
             self._client = ProductSearchClient(credentials=self._get_credentials())
         return self._client
 
+    @cached_property
```

Comment:
I'm trying to follow the conventions. get_conn also caches the API library instance.

I only introduced two improvements:

- I used property because it better expresses the purpose of this code. Creating getters and setters is not needed in Python. In my opinion, this is the impact of other languages that do not have support for this mechanism. 
- I created a decorator to separate logic.

I will add a comment.
> The access library creates many objects in the constructor. However, it is safe to use the same instance multiple times, so it has been retained for reuse.

I would like to point out that there are similar mechanisms in the project without comment:
https://github.com/PolideaInternal/airflow/blob/master/airflow/contrib/hooks/gcp_spanner_hook.py#L50-L52

Reply:
I looked at get_conn (I have not seen it before) and looked elsewhere, and I have second thought. I think it's just unnecessary here to use decorator as it introduces inconsistent coding. We are talking a lot about consistency on slack (I will still comment on that) but here we have now two ways of caching clients. The get_conn() is used across many hooks (and in some cases the same "if pattern" is used for caching). We cannot change it easily for all hooks. This seems to be common pattern. I would rather (for consistency) keep using the same pattern for image annotation rather than use decorator. Decorator in this case is used once and does not add the benefit of being reused  it makes it actually less readable in effect (in this particular part of code) because you have to look at what "cached_property" does. It would be different story if we introduced cached_property as a util and use it in a number of places consistently (then readability improves as you already know what cached_property does). Especially that it's next to simpler version of it.

Reply:
This now also introduces the operator `convert_http_exception` is unexpected. Now we have two ways to handle the requests:

- using decorator `convert_http_exception`
- using method `_handle_request` and lamda.

We agreed that this is a good solution, because it increases readability. The same argument applies here. 

I think that we should strive for a better future with small steps, and not look back all the time, if we see that the solution has a positive side, and only a consequence is against it. If we add a method, deleting it will be problematic.

Reply:
As discussed - let's put the decorator to a place where it will be directly reusable in the future (airflow/contrib/utils) and it will be fine then.

Reply:
I moved a decorator and added missing tests.